File: app/src/main/java/com/ilyap/yuta/utils/FileUtils.java
package com.ilyap.yuta.utils;

import android.content.Context;
import android.database.Cursor;
import android.net.Uri;
import android.provider.OpenableColumns;
import lombok.experimental.UtilityClass;

@UtilityClass
public class FileUtils {

    public static String getFileName(Context context, Uri uri) {
        if (uri == null) return null;

        String result = null;
        if (uri.getScheme().equals("content")) {
            try (Cursor cursor = context.getContentResolver().query(uri, new String[]{
                    OpenableColumns.DISPLAY_NAME
            }, null, null, null)) {
                if (cursor != null && cursor.moveToFirst()) {
                    int columnIndex = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME);
                    if (columnIndex != -1) {
                        result = cursor.getString(columnIndex);
                    }
                }
            }
        }
        return result;
    }
}

File: app/src/main/java/com/yuta/app/util/FileUtils.kt
package com.yuta.app.util

import android.content.Context
import android.net.Uri
import android.provider.OpenableColumns

object FileUtils {

    fun getFilename(context: Context, uri: Uri?): String? {
        return if (uri != null && uri.scheme == "content") {
            context.contentResolver.query(uri, arrayOf(OpenableColumns.DISPLAY_NAME), null, null, null)
                ?.use { cursor ->
                    if (cursor.moveToFirst()) {
                        val columnIndex = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME)
                        if (columnIndex != -1) cursor.getString(columnIndex) else null
                    } else null
                }
        } else null
    }
}
